aws_lambda_powertools/event_handler/api_gateway.py

```diff
@@ -226,12 +231,14 @@ def build(self, event: BaseProxyEvent, cors: Optional[CORSConfig] = None) -> Dic
             logger.debug("Encoding bytes response with base64")
             self.response.base64_encoded = True
             self.response.body = base64.b64encode(self.response.body).decode()
-        return {
+
+        payload = {
             "statusCode": self.response.status_code,
-            "headers": self.response.headers,
             "body": self.response.body,
             "isBase64Encoded": self.response.base64_encoded,
+            **event.header_serializer().serialize(headers=self.response.headers, cookies=self.response.cookies),
         }
+        return payload
```

Comment:
np: you could return it directly (no payload var), or are you doing to ease debugging in case `header_serializer()` goes haywire?

If that's the latter, you could pluck that out in a separate var then return the payload with the dict merge.

```python
serialized_headers = event.header_serializer().serialize(headers=self.response.headers, cookies=self.response.cookies)

return {
            "statusCode": self.response.status_code,
            "body": self.response.body,
            "isBase64Encoded": self.response.base64_encoded,
            **serialized_headers
}
```